content/200-concepts/100-components/01-prisma-schema/06-relations/425-referential-actions/100-cyclic-referential-actions.mdx

```diff
@@ -0,0 +1,111 @@
+---
+title: 'Cyclic referential actions with Microsoft SQL Server'
+metaTitle: 'Cyclic referential actions with Microsoft SQL Server'
+metaDescription: 'Circular references can occur between models when each model references another creating a closed loop. Since Microsoft SQL Server does not handle these situations out of the box, learn how to solve this problem.'
+tocDepth: 2
+---
+
+<TopBlock>
+
+Circular references can occur between models when each model references another, creating a closed loop. Most SQL databases handle these references out of the box. 
+
+With Microsoft SQL Server, however, if the [referential actions](./) on the relation field is set to something other than `NoAction`, the data model validator will throw an error the cyclic relation.
+
+This is dependent on whether or not the underlying scalar fields are set to be optional or required. If no referential action is specified on the relation, the default values are used:
+
+| Clause     | All of the scalar fields are optional | At least one scalar field is required |
+| :--------- | :----------------------- | :------------------------------------ |
+| `onDelete` | `SetNull`                | `NoAction`                            |
+| `onUpdate` | `Cascade`                | `Cascade`                             |
+
+</TopBlock>
+
+## Self relation
+
+The following model describes a self relation where the chicken references itself with fields that represent both the mother and father chickens.
+
+```prisma
+model Chicken {
+  id        Int      @id @default(autoincrement())
+  cock      Chicken? @relation(name: "a_self_relation")
+  hen       Chicken? @relation(name: "a_self_relation", fields: [chickenId], references: [id])
+  chickenId Int?
```

Comment:
I honestly do not understand this data model.

We have a `Chicken` table with many things in it. A chicken has a relation to another chicken, that is called `hen` and the backrelation is called `cock`. If the chicken we point to is the `hen`, why does that mean that the pointing cicken is the `cock`? Should it be the child instead?

Reply:
Because a chicken can have a relation to a hen and a cock (its parents)

Reply:
But that would require 2 relations, with two relation scalar fields (and also two back relations) if 1 chicken has 2 parents. Whatever is there right now only links a chicken to another chicken - and then just gives it different names.